test/rest_api/test_storage_service.py

```diff
@@ -40,8 +40,7 @@ def test_storage_service_keyspaces(cql, this_dc, rest_api):
         resp.raise_for_status()
         assert keyspace in resp.json()
 
-@pytest.mark.xfail(reason="rest_api suite doesn't support tablets yet (#17338), run test manually")
-def test_storage_service_keyspaces_replication(cql, this_dc, rest_api):
+def test_storage_service_keyspaces_replication(cql, this_dc, rest_api, skip_without_tablets):
```

Comment:
I'm not sure what I'm seeing here (I would have liked to see a comment explaining this, not just an answer here) - why do these tests need to be skipped without tablets? Didn't we have this REST API before we had tablets?

If with tablets we changed the format of the output (?) shouldn't we use the "has_tablets" flag to check the right value?

Or, if a REST API request responds differently depending on whether the specific keyspace has tablets (not according to the "--experimental" flag!), then on a Scylla supporting tablets we could test both with-tablets and without-tablets cases?

Reply:
Yes, I added these tests to cover tablet-related changes to these APIs. I ran these tests with `run` to verify that the added tablet functionality works. Then discovered that when running via `test.py`, this suite runs with vnodes and the test fails. So I added xfail to get the code in.